src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9333,6 +9333,12 @@ private ExpressionSyntax ParsePostFixExpression(ExpressionSyntax expr)
                         }
 
                         goto default;
+
+                    case SyntaxKind.ExclamationToken:
```

Comment:
>case [](start = 20, length = 4)

I gather the syntax is
``` antlr
primary-no-array-creation-expression
    : exclamation-expression
    ;
exclamation-expression
    : primary-expression '!'
    ;
```

In which case it must be a *semantic* error when you write `e!!`.

Reply:
I've added a commented out test for `!!` and will add the binding error in a subsequent PR.